checks/evaluation/binary_artifacts/binary_artifacts.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluation
+
+import (
+	"embed"
+	"path"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+//go:embed *.yml
+var rules embed.FS
+
+// BinaryArtifacts applies the score policy for the Binary-Artifacts check.
+func BinaryArtifacts(name string, dl checker.DetailLogger,
+	r *checker.BinaryArtifactData,
+) checker.CheckResult {
+	if r == nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	// Keep track of reported results.
+	reportedRuleResults := map[string]bool{
+		"BinaryGradleWrapperSafe":   false,
+		"BinaryArtifactsNotPresent": false,
+	}
+
+	// Apply the policy evaluation.
+	if r.Files == nil || len(r.Files) == 0 {
+		// Report findings for all rules.
+		if err := logDefaultFindings(dl, reportedRuleResults); err != nil {
+			return checker.CreateRuntimeErrorResult(name, err)
+		}
+		return checker.CreateMaxScoreResult(name, "no binaries found in the repo")
+	}
+
+	score := checker.MaxResultScore
+	for _, f := range r.Files {
+		//nolint:gosec
+		loc := checker.LocationFromFile(&f)
+		switch {
```

Comment:
Assuming there is only one file, a `gradle-wrapper.jar` file.
This code will only report a finding for the `BinaryGradleWrapperSafe` not the `BinaryArtifactsNotPresent` rule. 
I would expect findings for both?

Reply:
That'd good feedback. It's actually intentional. The idea is that all the binaries that are acceptable (maven, gradle, etc) will have a corresponding remediation (e.g., use the corresponding Action) and have their own rule; and we'll only report them under the `BinaryXXXWrapperSafe `

Can you take a look at the rule description and tell me if it's confusing?

Any ideas to remove this confusion besides description change or rule name changes? I'm open to anything :-)

NOTE: the idea is the following. Someone wants to write a policy. They can check that there are no failures under `BinaryArtifactsNotPresent` and none under `BinaryGradleWrapperSafe`. Otherwise they'd have to check for the filename in both rules, which is more complicated. This also makes rules independent so you dont need to link them to one another for policy.

Let's discuss more.